src/test/cs/Infinispan/HotRod/NearCacheFailoverTest.cs

```diff
@@ -0,0 +1,103 @@
+        {
+            this.servers = servers;
+        }
+    }
+
+    public class NearCacheFailoverTest
+    {
+        RemoteCacheManager remoteManager;
+        IMarshaller marshaller;
+        static StickyBalancingStragegy b;
+        HotRodServer server1;
+        HotRodServer server2;
+
+        FailOverRequestBalancingStrategyProducerDelegate d = delegate ()
+        {
+            b = new StickyBalancingStragegy();
+            return b;
+        };
+
+        [TestFixtureSetUp]
+        public void BeforeClass()
+        {
+            BeforeSuite();
+            ConfigurationBuilder conf = new ConfigurationBuilder();
+            conf.AddServer().Host("127.0.0.1").Port(11222);
+            conf.AddServer().Host("127.0.0.1").Port(11322)
+            .ConnectionTimeout(90000).SocketTimeout(6000)
+            .NearCache().Mode(NearCacheMode.INVALIDATED).MaxEntries(10);
+            conf.BalancingStrategyProducer(d);
+            marshaller = new JBasicMarshaller();
+            conf.Marshaller(marshaller);
+            remoteManager = new RemoteCacheManager(conf.Build(), true);
+        }
+
+        public void BeforeSuite()
+        {
+            server1 = new HotRodServer("clustered.xml");
+            server1.StartHotRodServer();
+            string jbossHome = System.Environment.GetEnvironmentVariable("JBOSS_HOME");
+            server2 = new HotRodServer("clustered.xml", "-Djboss.socket.binding.port-offset=100 -Djboss.server.data.dir=" + jbossHome + "/standalone/data100", 11322);
+            server2.StartHotRodServer();
+        }
+
+        [TestFixtureTearDown]
+        public void AfterClass()
+        {
+            remoteManager.Stop();
+            if (server1.IsRunning(2000))
+                server1.ShutDownHotrodServer();
+            if (server2.IsRunning(2000))
+                server2.ShutDownHotrodServer();
+        }
+
+        [Test]
+        public void NearCacheWithFailoverTest()
+        {
+            var cache = remoteManager.GetCache<string, string>();
+            cache.Clear();
+            ServerStatistics stats0 = cache.Stats();
+            cache.Put("key1", "value1");
+            cache.Get("key1");
+            var stats1 = cache.Stats();
+            Assert.AreEqual(int.Parse(stats0.GetStatsMap()["nearHits"]), int.Parse(stats1.GetStatsMap()["nearHits"]));
+            cache.Get("key1");         // near hit
+            var stats2 = cache.Stats();
+            Assert.AreEqual(int.Parse(stats1.GetStatsMap()["nearHits"]) + 1, int.Parse(stats2.GetStatsMap()["nearHits"]));
+            server2.ShutDownHotrodServer();
+            // After the shutdown the client failover on the 11222 server and reset the near cache
+            var stats3 = cache.Stats();
+            cache.Get("key1");         // remote hit
+            var stats4 = cache.Stats();
+            Assert.AreEqual(int.Parse(stats3.GetStatsMap()["nearHits"]), int.Parse(stats3.GetStatsMap()["nearHits"]));
```

Comment:
Here you compare stats3 with stats3. One of them should be stats4

Reply:
oh yes :)